src/include/duckdb/optimizer/join_order/query_graph.hpp

```diff
@@ -61,22 +46,23 @@ class QueryGraph {
 	string ToString() const;
 	void Print();
 
-	//! Create an edge in the edge_set
-	void CreateEdge(JoinRelationSet &left, JoinRelationSet &right, optional_ptr<FilterInfo> info);
 	//! Returns a connection if there is an edge that connects these two sets, or nullptr otherwise
-	vector<reference<NeighborInfo>> GetConnections(JoinRelationSet &node, JoinRelationSet &other);
+	const vector<reference<NeighborInfo>> GetConnections(JoinRelationSet &node, JoinRelationSet &other) const;
```

Comment:
Is this meant to be `const vector<reference<NeighborInfo>> &` (const reference) instead?
Otherwise it's not clear to me what the `const` modifier at the start is for

Reply:
first const is so that calling function can't change any of the elements in the vector (since they are references). The second const tells the compiler that the function doesn't modify any values of the `query_graph` class